src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,227 @@
+  private static final int TILE_SIZE = 125;
+  private static final int PADDING = 20;
+
+  private static final Comparator<MapVersion> mapVersionComparator = Comparator.nullsFirst(Comparator.comparing(MapVersion::size))
+                                                          .thenComparing(mapVersion -> mapVersion.map().displayName(),
+                                                                         String.CASE_INSENSITIVE_ORDER);
+
+  private static final Comparator<MatchmakerQueueMapPool> mapPoolComparator = Comparator.comparing(MatchmakerQueueMapPool::minRating, Comparator.nullsFirst(Double::compare))
+                                                                   .thenComparing(MatchmakerQueueMapPool::maxRating, Comparator.nullsLast(Double::compare));
+
+
+  private final MapService mapService;
+  private final UiService uiService;
+  private final PlayerService playerService;
+  private final FxApplicationThreadExecutor fxApplicationThreadExecutor;
+
+  public Pane root;
+  public FlowPane tilesContainer;
+  public ScrollPane scrollContainer;
+  public VBox loadingPane;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBracketsWithDuplicates;
+  private IntegerProperty playerBracketIndex = new SimpleIntegerProperty(0);
+
+  private DoubleProperty maxWidth = new SimpleDoubleProperty(0);
+
+  private DoubleProperty maxHeight = new SimpleDoubleProperty(0);
+
+
+  @Override
+  protected void onInitialize() {
+    JavaFxUtil.bindManagedToVisible(loadingPane);
+  }
+
+  @Override
+  public Pane getRoot() {
+    return root;
+  }
+
+  public double getMaxWidth(){
+    return this.maxWidth.get();
+  }
+  public void setMaxWidth(double value) {
+    this.maxWidth.set(value);
+  }
+  public DoubleProperty maxWidthProperty() {
+    return this.maxWidth;
+  }
+
+  public double getMaxHeight(){
+    return this.maxHeight.get();
+  }
+  public void setMaxHeight(double value) {
+    this.maxHeight.set(value);
+  }
+  public DoubleProperty maxHeightProperty() {
+    return this.maxHeight;
+  }
+
+
+  public void setQueue(MatchmakerQueueInfo queue) {
+    loadingPane.setVisible(true);
+    mapService.getMatchmakerBrackets(queue).subscribe(rawBrackets -> {
+      loadingPane.setVisible(false);
+
+      this.sortedBracketsWithDuplicates = this.getSortedBrackets(rawBrackets);
+
+      this.sortedBrackets = this.removeDuplicates(this.sortedBracketsWithDuplicates);
+
+      PlayerInfo player = playerService.getCurrentPlayer();
+      Integer rating = RatingUtil.getLeaderboardRating(player, queue.getLeaderboard());
+
+      this.playerBracketIndex.set(this.getPlayerBracketIndex(this.sortedBrackets, rating));
+
+      List<MapVersion> values = this.sortedBrackets.values().stream().flatMap(List::stream).toList();
+      this.resizeToContent(values.size(), TILE_SIZE);
+
+
+      fxApplicationThreadExecutor.execute(() -> values.forEach(this::addMapTile));
```

Comment:
So now here the sortedBrackets, sortedBrackets with duplicates, playerBracketIndex, and the values can be made into properties.

This can be done in the initialize method and you can utilize the map methods on properties to create observable values.

Then you can also add listeners to these properties that call resizeContent. That should cover the bases with regards to this method.